module/VuFind/src/VuFind/Controller/MyResearchController.php

```diff
@@ -733,7 +735,10 @@ public function profileAction()
                 if (' ** ' === $homeLibrary) {
                     $homeLibrary = null;
                 }
-                $user->changeHomeLibrary($homeLibrary);
+                // Update the home library and make sure library cards are kept in sync:
+                $user->setHomeLibrary($homeLibrary);
```

Comment:
This is only used in one place, so it seemed easier to expand the controller than to create a new method somewhere else to do this work.

Reply:
It is very little business logic, but does makes me wonder where to draw the line with business logic in controllers?

Reply:
Yeah, on further reflection, I really don't like this. I've decided to move the logic into the ILSAuthenticator. That's not a perfect solution either, but it's at least better encapsulation. If you think it's worth the extra effort, I could create a `HomeLibraryUpdateInterface` (or something like that) and make it an alias of the ILSAuthenticator so that we could potentially factor these things apart in future if we wanted to. (I'm not going to delay merging this for that change, but I'm happy to open a follow-up PR if you think it's worthwhile).

Reply:
(Though the more I think about it, I think this is probably justifiable as part of the ILSAuthenticator, since in some situations the user's choice of home library could be related to authentication). Not saying I'm 100% comfortable with it, but I don't think it's a crime. :-)